dbusmock/gtest.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/python3
+
+import os
+import sys
+import subprocess
+import functools
+from future.utils import with_metaclass
+
+if sys.version_info[0] < 3:
+    PIPE_DEVNULL = open(os.devnull, 'wb')
+else:
+    PIPE_DEVNULL = subprocess.DEVNULL
+
+
+class _GTestSingleProp(object):
+    """Property which creates a bound method for calling the specified test."""
+    def __init__(self, test):
+        self.test = test
+
+    @staticmethod
+    def __func(self, test):
+        self._gtest_single(test)
+
+    def __get__(self, obj, cls):
+        bound_method = self.__func.__get__(obj, cls)
+        partial_method = functools.partial(bound_method, self.test)
+        partial_method.__doc__ = bound_method.__doc__
+
+        return partial_method
+
+
+class _GTestMeta(type):
+    def __new__(cls, name, bases, namespace, **kwds):
+        result = type.__new__(cls, name, bases, dict(namespace))
+
+        if result.g_test_exe is not None:
+            try:
+                _GTestMeta.make_tests(result.g_test_exe, result)
+            except Exception as e:
+                print('')
+                print(e)
+                print('Error generating separate test funcs, will call binary once.')
+                result.test_all = result._gtest_all
+
+        return result
+
+    @staticmethod
+    def make_tests(exe, result):
+        test = subprocess.Popen([exe, '-l'], stdout=subprocess.PIPE, stderr=PIPE_DEVNULL)
```

Comment:
We ran into the problem today that this particular execution is done without an X11 server started, which means that it can fail if the process needs X11 to initialise.